service/storage_service.cc

```diff
@@ -4869,6 +4874,18 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
            // Return an error since the command is from outdated leader
            co_return result;
         }
+
+        {
```

Comment:
to restrict the scope of the `state` local variable

Reply:
We don't need to do it for each and every variable, what a strange desire. The point is to limit a block of code that performs one logically isolated task, so that it does not leave behind garbage - used and no longer needed variables. If the containing function is large, these variables may accidentally conflict with some other variables below and cause unnecessary questions. This is the same reason why `if(type var = init(); cond(var)) { use(var); }` syntax was introduced, but it's not convenient to use it here.

Reply:
> We don't need to do it for each and every variable, what a strange desire
Why not? Why this one is special?

> The point is to limit a block of code that performs one logically isolated task

If the work worth to be isolated it should be in a function.

> cause unnecessary questions

What kind of questions? There will be an error about variable re-declaration. It may mean that the name of the variable was not descriptive enough.

About the 'if' you do not introduce a scope in the middle of nowhere. You have a scope and it is of course fine to limit a variable to it, but adding scopes in random places does not help. On the contrary.

Reply:
> > If the work worth to be isolated it should be in a function.
> 
> No, is depends. The function by itself adds cognitive load, if the code is used only in one place it's more often doesn't deserve to be extracted.

Good named function reduces cognitive load. The code that is used in one place does not deserve global function, I agree. But C++ have lambdas to simulate local functions easily.

> 
> > What kind of questions? There will be an error about variable re-declaration.
> 
> Why am I getting this error?

If you declare the same variable twice you get an error about re-declaration.

> 
> > It may mean that the name of the variable was not descriptive enough.
> 
> Braces allows use not to invent artificial names for things that are basically the same, but used in slightly different contexts in the same function. Here's the typical example: https://github.com/gusev-p/scylla/blob/095f35a47d4dafe5f08f8a76c1fb49b16c2b7592/test/boost/token_metadata_test.cc#L55
> 
> We have several cases, each case lives in its own block, we don't need to invent a name for `erm` and can use the same code structure for both of them.
> 
> I don't see what's wrong with these braces.

I do not like them for three reasons: take vertical space, take horizontal space, make one think why they are there (cognitive load at you put it). For cases where you need to limit a guard lifetime or want to destroy unneeded large object in a sleeping function such artificial scope has a lot of sense, so when I see it I start looking for those reasons.

Anyway this is a taste thing. No need to block the series because of it.

Reply:
To me these local scopes reduce cognitive load, because they make lifetimes of variables smaller. I know the variables declared in the scope won't be used later.

Reply:
If a function is so long that it matters may be it should be split into several functions. But we discussed that already. I do not think we can come up with new arguments.